src/transaction_builder.js

```diff
@@ -721,20 +721,27 @@ export class TransactionBuilder {
     networkPassphrase
   ) {
     const innerOps = innerTx.operations.length;
-    const innerBaseFeeRate = new BigNumber(innerTx.fee).div(innerOps);
-    const base = new BigNumber(baseFee);
+    let inclusionFee = new BigNumber(innerTx.fee).div(innerOps);
 
-    // The fee rate for fee bump is at least the fee rate of the inner transaction
-    if (base.lt(innerBaseFeeRate)) {
-      throw new Error(
-        `Invalid baseFee, it should be at least ${innerBaseFeeRate} stroops.`
-      );
+    // Do we need to do special Soroban fee handling? We only want the fee-bump
+    // requirement to match the inclusion fee, not the inclusion+resource fee.
+    const env = innerTx.toEnvelope();
+    switch (env.switch().value) {
+      case xdr.EnvelopeType.envelopeTypeTx().value:
+        const sorobanData = env.v1().tx().ext().value();
+        inclusionFee -= sorobanData?.resourceFee() ?? 0;
+        break;
     }
 
+    const base = new BigNumber(baseFee);
     const minBaseFee = new BigNumber(BASE_FEE);
 
-    // The fee rate is at least the minimum fee
-    if (base.lt(minBaseFee)) {
+    // The fee rate for fee bump is at least the fee rate of the inner transaction
+    if (base.lt(inclusionFee)) {
```

Comment:
I'm not really familiar with the math here, so just wanted to double-check with you: are you going to correctly rebuild the outer tx fee given the inclusion fee? Specifically you need to set it to
`outerFee = inclusionFee * (innerOps + 1) + sorobanResourceFee`.

Reply:
> the inner tx may or may not have its fee set to include the resource fee

Yeah, inner tx `fee` is basically meaningless (see the comment above).

> the fee tx must include the resource fee

Yes, the outer tx fee is what is actually going to be used in all the fee-related computations.